apps/doji-frontend/src/pages/schedule-session/index.tsx

```diff
@@ -0,0 +1,268 @@
+                      onChange={onChange}
+                      renderInput={(params) => (
+                        <TextField
+                          {...params}
+                          error={!!errors.startTime}
+                          helperText={errors.startTime?.message}
+                        />
+                      )}
+                    />
+                  )}
+                ></Controller>
+              </LocalizationProvider>
+            </Grid>
+            <Grid item xs={6} marginBottom={3}>
+              <LocalizationProvider dateAdapter={AdapterDateFns}>
+                <Controller
+                  name="endTime"
+                  control={control}
+                  render={({ field: { onChange, value } }) => (
+                    <TimePicker
+                      {...register('endTime')}
+                      label="End Time"
+                      value={value}
+                      onChange={onChange}
+                      renderInput={(params) => (
+                        <TextField
+                          {...params}
+                          error={!!errors.endTime}
+                          helperText={errors.endTime?.message}
+                        />
+                      )}
+                    />
+                  )}
+                ></Controller>
+              </LocalizationProvider>
+            </Grid>
+            <Grid item xs={12} marginBottom={2}>
+              <Typography variant="large" fontWeight={700}>
+                Participants
+              </Typography>
+            </Grid>
+            <Grid item xs={12}>
+              <Controller
+                name="participants"
+                control={control}
+                render={({ field: { onChange, value } }) => <TagsInput onChange={onChange} />}
+              ></Controller>
+            </Grid>
+          </Grid>
+          <Box position="relative" bottom={0} left={0} right={0} marginTop={3}>
+            <Container sx={{ padding: 2, backgroundColor: 'white' }} maxWidth="sm">
+              <Grid container alignItems="center">
+                <Grid item xs={6}>
+                  <Typography variant="large" fontWeight={700}>
+                    Total Price
+                  </Typography>
+                </Grid>
+                <Grid item xs={6} textAlign="right">
+                  <Typography variant="large" fontWeight={700} color="#367D7F">
+                    {calculateTotal()}
+                  </Typography>
+                  <Typography variant="regular" fontWeight={400} color="#367D7F">
+                    &nbsp; Doji coins
+                  </Typography>
+                </Grid>
+                <Grid item xs={12} marginTop={2} marginBottom={1}>
+                  <Button fullWidth type="submit">
+                    Schedule
+                  </Button>
+                </Grid>
+              </Grid>
+            </Container>
+          </Box>
+        </form>
+      </Box>
```

Comment:
can you extract the code? create a new subcomponent

Reply:
It is the component that depend on the state of the page and form data (total price calculation). I think it cannot be extracted.

Reply:
You can pass the state and the form data as `props` to the component like

```tsx
function ComponentA() {
  const [state, setState] = useState( ... )
  const { control } = useForm( ... )

  return <ComponentB control={control} state={state} /> 
}

function ComponentB({ control, state}) {
  const watchAll = useWatch({ control })  
  return <>{state}</>
}
``` 

ปล. สำหรับ watchAll ถ้าทำท่านี้ เวลาเรา input form มันจะ rerender แค่ componentB